src/Client/Client.php

```diff
@@ -55,13 +55,18 @@ public function createRequest(Request $request)
      *
      * @param \Assertis\Http\Request\Request $request
      * @return ResponseInterface
+     * @throws Exception
      */
     public function send(Request $request)
     {
         try {
             $response = $this->guzzleClient->send($this->createRequest($request));
-        } catch (RequestException $exception) {
-            $response = $exception->getResponse();
+        } catch (Exception $exception) {
+            if ($exception instanceof RequestException) {
+                $response = $exception->getResponse();
```

Comment:
It's minor, but you can just return here and above in the try...catch block, and skip the else... block